src/ng/templateRequest.js

```diff
@@ -68,8 +68,14 @@ function $TemplateRequestProvider() {
       // are included in there. This also makes Angular accept any script
       // directive, no matter its name. However, we still need to unwrap trusted
       // types.
-      if (!isString(tpl) || !$templateCache.get(tpl)) {
```

Comment:
The reasoning is that people asked for wrong fixes several times, so my original code was probably confusing. I've unrolled the if to accomodate clearer comments, but that can be reverted if you prefer.